e2e/aws_test.go

```diff
@@ -0,0 +1,171 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: framework.CAPINamespace,
+		},
+		Spec: awsv1.AWSClusterSpec{
+			Region: mapiProviderSpec.Placement.Region,
+		},
+	}
+
+	if err := cl.Create(ctx, awsCluster); err != nil && !apierrors.IsAlreadyExists(err) {
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	Eventually(func() (bool, error) {
+		patchedAWSCluster := &awsv1.AWSCluster{}
+		err := cl.Get(ctx, client.ObjectKeyFromObject(awsCluster), patchedAWSCluster)
+		if err != nil {
+			return false, err
+		}
+
+		if patchedAWSCluster.Annotations == nil {
+			return false, nil
+		}
+
+		if _, ok := patchedAWSCluster.Annotations[clusterv1.ManagedByAnnotation]; !ok {
+			return false, nil
+		}
+
+		return patchedAWSCluster.Status.Ready, nil
+	}, framework.WaitShort).Should(BeTrue())
+
+	return awsCluster
+}
+
+func createAWSMachineTemplate(cl client.Client, mapiProviderSpec *mapiv1.AWSMachineProviderConfig) *awsv1.AWSMachineTemplate {
+	By("Creating AWS machine template")
+
+	Expect(mapiProviderSpec).ToNot(BeNil())
+	Expect(mapiProviderSpec.IAMInstanceProfile).ToNot(BeNil())
+	Expect(mapiProviderSpec.IAMInstanceProfile.ID).ToNot(BeNil())
+	Expect(mapiProviderSpec.InstanceType).ToNot(BeEmpty())
+	Expect(mapiProviderSpec.Placement.AvailabilityZone).ToNot(BeEmpty())
+	Expect(mapiProviderSpec.AMI.ID).ToNot(BeNil())
+	Expect(mapiProviderSpec.Subnet.Filters).ToNot(HaveLen(0))
+	Expect(mapiProviderSpec.Subnet.Filters[0].Values).ToNot(HaveLen(0))
+	Expect(mapiProviderSpec.SecurityGroups).ToNot(HaveLen(0))
+	Expect(mapiProviderSpec.SecurityGroups[0].Filters).ToNot(HaveLen(0))
+	Expect(mapiProviderSpec.SecurityGroups[0].Filters[0].Values).ToNot(HaveLen(0))
+
+	uncompressedUserData := true
+	awsMachineTemplate := &awsv1.AWSMachineTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      awsMachineTemplatenName,
+			Namespace: framework.CAPINamespace,
+		},
+		Spec: awsv1.AWSMachineTemplateSpec{
+			Template: awsv1.AWSMachineTemplateResource{
+				Spec: awsv1.AWSMachineSpec{
+					UncompressedUserData: &uncompressedUserData,
+					IAMInstanceProfile:   *mapiProviderSpec.IAMInstanceProfile.ID,
+					InstanceType:         mapiProviderSpec.InstanceType,
+					FailureDomain:        &mapiProviderSpec.Placement.AvailabilityZone,
+					CloudInit: awsv1.CloudInit{
+						InsecureSkipSecretsManager: true,
+					},
+					AMI: awsv1.AMIReference{
+						ID: mapiProviderSpec.AMI.ID,
+					},
+					Subnet: &awsv1.AWSResourceReference{
+						Filters: []awsv1.Filter{
+							{
+								Name:   "tag:Name",
+								Values: mapiProviderSpec.Subnet.Filters[0].Values,
+							},
+						},
+					},
+					AdditionalSecurityGroups: []awsv1.AWSResourceReference{
+						{
+							Filters: []awsv1.Filter{
+								{
+									Name:   "tag:Name",
+									Values: mapiProviderSpec.SecurityGroups[0].Filters[0].Values,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	Expect(cl.Create(ctx, awsMachineTemplate)).To(Succeed())
```

Comment:
Is this object being cleaned up between tests?

Reply:
yes, I think we should clean it up in case we would like to have different templates for tests